src/pages/studyView/table/GeneTable.tsx

```diff
@@ -416,73 +447,40 @@ export class GeneTable extends React.Component<GeneTableProps, {}> {
             (row: GeneTableUserSelectionWithIndex) => row.uniqueKey === uniqueKey
         );
         if (_.isUndefined(record)) {
-            let dataIndex = -1;
-            // definitely there is a match
-            const datum = _.find(
-                this.tableData,
-                (row, index: number) => {
-                    const exist = row.uniqueKey! === uniqueKey;
-                    if (exist) {
-                        dataIndex = index;
-                    }
-                    return exist;
-                }
-            );
-
-            if (!_.isUndefined(datum)) {
-                this.preSelectedRows.push({
-                    rowIndex: dataIndex,
-                    uniqueKey: datum.uniqueKey!,
-                });
-            }
+            this.preSelectedRows.push({ uniqueKey: uniqueKey});
         } else {
-            this.preSelectedRows = _.xorBy(this.preSelectedRows, [record], "rowIndex");
+            this.preSelectedRows = _.xorBy(this.preSelectedRows, [record], "uniqueKey");
         }
     }
 
     @autobind
     @action
     afterSelectingRows() {
+        console.log(JSON.stringify(this.preSelectedRows))
         this.props.onUserSelection(
             this.preSelectedRows.map(row => row.uniqueKey)
         );
         this.preSelectedRows = [];
     }
 
-    @computed
-    get selectedRows() {
-        if (this.props.filters.length === 0) {
-            return [];
-        } else {
-            return _.reduce(
-                this.tableData,
-                (
-                    acc: GeneTableUserSelectionWithIndex[],
-                    row,
-                    index: number
-                ) => {
-                    if (_.includes(this.props.filters, row.uniqueKey)) {
-                        acc.push({
-                            rowIndex: index,
-                            uniqueKey: row.uniqueKey
-                        });
-                    }
-                    return acc;
-                },
-                []
-            );
-        }
-    }
-
     @autobind
     isSelectedRow(data: GeneTableRow) {
         return !_.isUndefined(
-            _.find(_.union(this.selectedRows, this.preSelectedRows), function (row) {
+            _.find(_.union(this.selectedRowsIndexes, this.preSelectedRows), function (row) {
```

Comment:
this is a linear search through rows, which will run every time isSelectedRow is called, which I'm assuming is once per row to check how to render it? this means quadratic time, which is really slow for long lists. it's better to make selectedRowsIndexes/Keys be a set so that we can check membership in constant time.

Reply:
also, `_.union(this.selectedRowsIndexes, this.preSelectedRows)` doesn't need to be recomputed for each call -  putting that computation into a `@computed` would make it cached and reusable.